cmd/estuary-shuttle/ping.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"context"
+	"sort"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/peerstore"
+	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
+	"github.com/multiformats/go-multiaddr"
+)
+
+type Libp2pHost struct {
```

Comment:
do you know where we're going to be constructing this shuttle-side yet? wondering if there's an existing struct that might be more convenient to use instead of us having to construct this on the caller. something like `AddrInfo` but seems like not exactly that

Reply:
hmm, these two variables are present together in the [MinerInfo](https://github.dev/filecoin-project/lotus/blob/master/api/types.go#L292) struct, but it has a lot of other stuff too so it's kinda heavy. 

I can't see anywhere where the `peerID` and `MultiAddress` are sent to the shuttle via RPC, so we'd be adding something new for that anyways. Maybe we should define this struct elsewhere so it's more generally accessible in the Shuttle package?